algebird-core/src/main/scala/com/twitter/algebird/BloomFilter.scala

```diff
@@ -150,6 +154,8 @@ case class BFItem(item: String, hashes: BFHash, width: Int) extends BF {
 
   def +(other: String) = this ++ BFItem(other, hashes, width)
 
+  def checkAndAdd(other: String): (BF, ApproximateBoolean) = (this + other, ApproximateBoolean.exact(other == item))
```

Comment:
if we are checking equality here anyway, perhaps we should not use + if other == item?